independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Beans.java

```diff
@@ -119,6 +126,23 @@ static BeanInfo createClassBean(ClassInfo beanClass, BeanDeployment beanDeployme
         return bean;
     }
 
+    private static ScopeInfo inheritScope(ClassInfo beanClass, BeanDeployment beanDeployment) {
+        Type superClass = beanClass.superClassType();
+        while (!superClass.equals(Type.create(OBJECT_NAME, Kind.CLASS))) {
+            for (AnnotationInstance annotation : beanDeployment.getIndex().getClassByName(superClass.name())
```

Comment:
`beanDeployment.getIndex().getClassByName()` could return `null` if the class is not available.

Reply:
Hmm, how can it happen that I use the index to identify the superclass yet the actual superclass isn't in the index?
I'll change it (+1 for defensive code) but just wonder how that happens.

Reply:
The index could only contain the name of the class not the `ClassInfo` itself. For example if your class extends a class from a different jar that is not indexed by Quarkus.

Reply:
I see, makes sense, thanks!